src/override/malloc.cc

```diff
@@ -116,6 +116,61 @@ extern "C"
   }
 #endif
 
+#if !defined(SNMALLOC_NO_REALLOCARR)
+  SNMALLOC_EXPORT int
+    SNMALLOC_NAME_MANGLE(reallocarr)(void* ptr_, size_t nmemb, size_t size)
+  {
+    int err = errno, r;
+    auto& a = ThreadAlloc::get();
+    bool overflow = false;
+    size_t sz = bits::umul(size, nmemb, overflow);
+    if (sz == 0)
+    {
+      errno = err;
+      return 0;
+    }
```

Comment:
I would need to catch it later with 0 alloc the pointer is null.

Reply:
So currently we only return nullptr if we are out of memory. So we could check with an assert, and make it such that a test would fail, if we changed that behaviour.  However, your code is much less fragile.  So lets keep it.